stablehlo/dialect/StablehloOps.cpp

```diff
@@ -1517,11 +1517,36 @@ DenseI64ArrayAttr getBroadcastDimensionsFromBroadcastSizes(
       getBroadcastDimensionsFromBroadcast(broadcastSizesSize, operandRank));
 }
 
+namespace {
+bool haveSimpleCompatibleDimentions(RankedTensorType operand,
+                                    RankedTensorType result) {
+  auto operandTy = cast<ShapedType>(operand);
+  auto resultTy = cast<ShapedType>(result);
+  ArrayRef<int64_t> operandShape = operandTy.getShape();
+  ArrayRef<int64_t> resultShape = resultTy.getShape();
+  // Two dimensions are compatible when:
+  // - they are equal, or
+  // - one of them is 1.
+  // For the simple broadcast check that we don't have to extend dimentions -
+  // reject the case when only one of them is 1.
```

Comment:
Can we move this above the function, this will help understand what "simple compatible dimensions" are, and give an example of "simple compatible" and "simple not compatible" dimensions to make this easier to understand for future reviewers/contributors

Reply:
PTAL

Reply:
LGTM! thanks! There's more intricacies to broadcast op than I'd thought.